pkg/storage/cloud/cloud_io.go

```diff
@@ -172,6 +184,7 @@ func (r *ResumingReader) Open() error {
 // Read implements io.Reader.
 func (r *ResumingReader) Read(p []byte) (int, error) {
 	var lastErr error
+	span := tracing.SpanFromContext(r.Ctx)
```

Comment:
`Read` can be a pretty hot path. I guess this is *external* storage so maybe it is ok?

Reply:
can't we just move this line to below, inside the error case?

Reply:
misunderstood your comment, done.